app/paid/paid_test.go

```diff
@@ -0,0 +1,74 @@
+package paid
+
+import (
+	"encoding/pem"
+	"fmt"
+	mrand "math/rand"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var testSDHash = "bea4d30a1868a00e98297cfe8cdefc1be6c141b54bea3b7c95b34a66786c22ab4e9f35ae19aa453b3630e76afbd24fe2"
+var testTxID = "e5a6a9ef4433b8868cebb770d1d2244eed410eb7503a9ffa82dce40aa62bbae7"
+
+func generateKeyFile() (string, error) {
+	keyFilename := fmt.Sprintf("test_private_key_%v.pem", time.Now().Unix())
+	cmd := fmt.Sprintf("openssl genrsa -out %s 2048", keyFilename)
+	if _, err := exec.Command("bash", "-c", cmd).Output(); err != nil {
+		return "", fmt.Errorf("command %v failed: %v", cmd, err)
+	}
+	return keyFilename, nil
+}
+
+func TestMain(m *testing.M) {
+	keyFile, err := generateKeyFile()
+	if err != nil {
+		panic(err)
+	}
+
+	err = InitPrivateKey(keyFile)
+	if err != nil {
+		panic(err)
+	}
+
+	pubKM = &pubKeyManager{}
+	err = pubKM.loadFromBytes(km.PublicKeyBytes())
+	if err != nil {
+		panic(err)
+	}
+
+	code := m.Run()
+	os.Remove(keyFile) // defers do not run after os.Exit()
```

Comment:
this is another thing `io.Reader` solves

Reply:
went with []byte